addon/src/modifiers/inputmask.js

```diff
@@ -1,39 +1,49 @@
 /* globals Inputmask */
 import Modifier from 'ember-modifier';
 import { getOwner } from '@ember/application';
+import InputMask from "inputmask";
+import { registerDestructor } from '@ember/destroyable';
+
+let setDefaults = undefined;
 
 export default class InputmaskModifier extends Modifier {
-  getArgs() {
-    return Object.keys(this.args.named).length
-      ? this.args.named
-      : this.args.positional[0] || {};
+  constructor(owner, args) {
+    super(owner, args);
+    registerDestructor(this, this.cleanup);
   }
 
-  didUpdateArguments() {
-    this._setInputMask();
+  cleanup = () => {
+    if (this.element.inputmask) {
+      this.element.inputmask.remove();
+    }
   }
 
-  didInstall() {
-    import('inputmask/dist/inputmask').then(() => {
-      this._initialize();
-    });
+  getArgs(positional, named) {
+    return Object.keys(named).length
+      ? named
+      : positional[0] || {};
   }
 
-  willRemove() {
-    if (this.element.inputmask) {
-      this.element.inputmask.remove();
+  element;
+
+  modify(element, positional, named) {
+    if (!setDefaults) {
+      this._setDefaults();
+      this.setDefaults = true;
     }
-  }
 
-  _initialize() {
-    this._setDefaults();
+    this.element = element;
 
-    const args = this.getArgs();
+    const args = this.getArgs(positional, named);
 
-    this._setInputMask(args);
+    new InputMask(args).mask(element);
```

Comment:
Looks like you removed _setInputMask invoke here but method still exists. I believe we should revert this line back.

Reply:
I meant to delete the method. Its a one line method that is invoke in only one place now. Also no longer needs to return.

Its your call. I think deleting the method is the way to go. You want me to add back the invoke?

Reply:
Nope I meant if you want to keep it we should revert this line back. If you want to remove totally fine.